src/scalar/scalar.ts

```diff
@@ -9,3 +15,20 @@ export interface Scalar<T> {
 }
 
 export type Vector = Scalar<unknown>[];
+
+export const newScalar = (dataType: DataType): Scalar<unknown> => {
+  if (DataType.isBool(dataType)) {
+    return new Bool(false);
```

Comment:
Can't we pass undefined/null instead of zero values and init the scalars as invalid?

Reply:
Yes we can, I'll update the classes